machine/clients/slack.py

```diff
@@ -93,6 +94,80 @@ async def _process_users_channels(self, _: AsyncBaseSocketModeClient, req: Socke
             elif event["type"] == "member_joined_channel":
                 await self._on_member_joined_channel(event)
 
+    async def fetch_all_users(client):
+        all_users = []
+        cursor = None
+
+        while True:
+            try:
+                # Fetch the users list with pagination
+                response = await client.web_client.users_list(limit=5000, cursor=cursor)
+                all_users += response["members"]
+                cursor = response.get("response_metadata", {}).get("next_cursor")
+
+                logger.info(f"{len(all_users)} users loaded!")
+
+                if not cursor:
+                    break
+
+            except SlackApiError as e:
+                # Handle rate limiting
+                if e.response is 429:
+                    retry_after = int(e.response.headers.get("Retry-After", 1))
+                    logger.warning(f"Rate limit hit. Retrying after {retry_after} seconds...")
+                    await asyncio.sleep(retry_after)
+                else:
+                    # Handle other Slack API errors
+                    logger.error(f"Error fetching channels: {e.response['error']}")
+                    break
+
+            except Exception as e:
+                # Handle any other exceptions
+                logger.error(f"Unexpected error: {e}")
+                break
+
+        return all_users
+
+    async def fetch_all_channels(client) -> List[Dict[str, Any]]:
```

Comment:
You can use lowercase `list` and `dict` here. In more recent versions of Python this is the recommended way